### DISCUSSION TITLE
Elevation with Dem 30

### DISCUSSION THREAD

**Discussion by libertykite** [open-meteo/open-meteo#211] [General]
Hi Patrick,
First of all a big thank you for your work, it's both pretty crazy and really very very useful for my project!

As I had put in a message, I have a problem with the api elevation and I am trying to understand why.

I have a hundred beaches in South Africa for which I would like to know the altitude and the api really returns completely false values.
So, I tried to install open-meteo locally and I downloaded the dem-90 data, I get the same inconsistencies.

Now I am downloading the Dem-30 data to see if I gain accuracy in this area of the globe. And I had a question, do I have to run the same command to convert them?

swift run openmeteo-api download-dem dem-30 ?

**Comment by patrick-zippenfenig** [comment]
Unfortunately, that does not work ;-) DEM-30 data is not supported by my code. According to their documentation it contains even more missing data than DEM-90.

I am sorry I did not yet fix your reported issue in https://github.com/open-meteo/open-meteo/issues/185. I checked it initially and did not spot any issues in my code, but the Copernicus data explorer somehow shows correct elevation data for South Africa.

In the next days I will check again. Maybe I can spot the issue now.

If it does not work, I will implement support for DEM-30

**Comment by libertykite** [comment]
I just did a lot of tests for different beaches (South Africa, Madagascar, Australia) and every time it's wrong.
I also tested the countries of the North (Norway, Iceland,..) and there it is always correct.

I wonder if for the southern countries, you don't have a one tile offset or something like that?

**Reply by patrick-zippenfenig** [reply]
It could have been an error with a negative modulo calculation and then used a wrong tile offset. I applied a fix in https://github.com/open-meteo/open-meteo/commit/b9828cfdf7225c494f01e6dfaf27be08b8cb109b. My unit tests had only smaller differences, but the beach location seems now mostly ok (25 meters instead of maybe 5).

The fixed version is deployed to production. Could you check again? Thanks!
